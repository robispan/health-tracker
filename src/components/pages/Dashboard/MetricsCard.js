import React from 'react';

import { CircularProgressbar, buildStyles } from 'react-circular-progressbar';
import 'react-circular-progressbar/dist/styles.css';

const MetricsCard = ({ ratio, label }) => {
  return (
    <div className='metrics-card'>
      <h3 className='metrics-card-title center'>{label}</h3>
      {ratio ? (
        <CircularProgressbar
          value={ratio}
          text={`${ratio}%`}
          background
          backgroundPadding={6}
          styles={buildStyles({
            backgroundColor: '#00b9bb',
            textColor: '#fff',
            pathColor: '#fff',
            trailColor: 'transparent'
          })}
        />
      ) : (
        <div className='center'>
          Connect trackers on homepage to see your progression!
        </div>
      )}
    </div>
  );
};

export default MetricsCard;
