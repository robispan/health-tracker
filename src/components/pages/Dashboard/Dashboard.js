import React, { useContext } from 'react';

import StravaContext from '../../../context/strava/stravaContext';

import MetricsCard from './MetricsCard';

const Dashboard = () => {
  const { stravaData } = useContext(StravaContext);

  const targetMetrics = {
    targetWeight: 50,
    heartRate: 55
  };

  const weightRatio =
    stravaData && stravaData.weight
      ? Math.round((100 * targetMetrics.targetWeight) / stravaData.weight)
      : null;

  return (
    <div>
      <h2 className='center upper'>Dashboard</h2>
      <div className='grid'>
        {/* Weight */}
        <MetricsCard label='Weight loss progression' ratio={weightRatio} />

        {/* Heart rate */}
        <MetricsCard label='Heart rate progression' />
      </div>
    </div>
  );
};

export default Dashboard;
