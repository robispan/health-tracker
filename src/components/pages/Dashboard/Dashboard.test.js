import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import StravaContext from '../../../context/strava/stravaContext';

import Dashboard from './Dashboard';

afterEach(cleanup);

test('renders correctly without context data', () => {
  const { asFragment } = render(
    <StravaContext.Provider value={{ stravaData: null }}>
      <Dashboard />
    </StravaContext.Provider>
  );
  expect(asFragment()).toMatchSnapshot();
});

test('renders correctly with context data', () => {
  const { asFragment } = render(
    <StravaContext.Provider value={{}}>
      <Dashboard />
    </StravaContext.Provider>
  );
  expect(asFragment()).toMatchSnapshot();
});
