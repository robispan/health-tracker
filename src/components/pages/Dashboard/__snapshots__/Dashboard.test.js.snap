// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with context data 1`] = `
<DocumentFragment>
  <div>
    <h2
      class="center upper"
    >
      Dashboard
    </h2>
    <div
      class="grid"
    >
      <div
        class="metrics-card"
      >
        <h3
          class="metrics-card-title center"
        >
          Weight loss progression
        </h3>
        <div
          class="center"
        >
          Connect trackers on homepage to see your progression!
        </div>
      </div>
      <div
        class="metrics-card"
      >
        <h3
          class="metrics-card-title center"
        >
          Heart rate progression
        </h3>
        <div
          class="center"
        >
          Connect trackers on homepage to see your progression!
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`renders correctly without context data 1`] = `
<DocumentFragment>
  <div>
    <h2
      class="center upper"
    >
      Dashboard
    </h2>
    <div
      class="grid"
    >
      <div
        class="metrics-card"
      >
        <h3
          class="metrics-card-title center"
        >
          Weight loss progression
        </h3>
        <div
          class="center"
        >
          Connect trackers on homepage to see your progression!
        </div>
      </div>
      <div
        class="metrics-card"
      >
        <h3
          class="metrics-card-title center"
        >
          Heart rate progression
        </h3>
        <div
          class="center"
        >
          Connect trackers on homepage to see your progression!
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
